### PR TITLE
BUG: DataFrame.to_records() bug in converting datetime64 index

### PR DESCRIPTION
**Pull request by amolkahat**
BUG: DataFrame.to_records() bug in converting datetime64 index with timezone #13937

### CODE REVIEW

**Comment by jreback** [comment]
needs tests and a whats new note.

**Comment by amolkahat** [comment]
@jreback  Tests for this issue is already present in file pandas/pandas/tests/frame/test_convert_to.py.

**Comment by jreback** [comment]
if there were tests then your change would fail. pls add them

**Comment by jreback** [comment]
there are very minimal tests in any event
and certainly nothing related to conversions of tz

**Comment by amolkahat** [comment]
@jreback This build with python2.7 failed. But there is not an error with the tests, there is an error with the Travis CI. It shows error like "The command "ci/lint.sh" exited with 1."

**Comment by jreback** [comment]
you need to lint the code.

**Comment by amolkahat** [comment]
@jreback Build is successful. Please take a look.

File: pandas/tests/frame/test_convert_to.py [hunk 1]
```diff
@@ -179,3 +178,31 @@ def test_to_records_with_unicode_index(self):
             .to_records()
         expected = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
         tm.assert_almost_equal(result, expected)
+
+    def test_to_records_with_tz_gmt(self):
+        data = [datetime.datetime.now(pytz.timezone('UTC')) for _ in range(10)]
+        df = DataFrame({'datetime': data})
+        df.set_index('datetime', inplace=True)
+        dfgmt = df.tz_convert("GMT")
+        self.assertNotEqual(df.to_records()['datetime'][0].tzinfo,
```
Comment (jreback) [inline pandas/tests/frame/test_convert_to.py]: construct the expected frame and use ``assert_frame_equal`` (and same below)

**Pull request closed without merging**